api/nodes/serializers.py

```diff
@@ -375,3 +377,61 @@ class Meta:
     @staticmethod
     def get_id(obj):
         return '{}:{}'.format(obj.node._id, obj.provider)
+
+class NodeAlternativeCitationSerializer(JSONAPISerializer):
+
+    id = IDField(source="_id", read_only=True)
+    type = TypeField()
+    name = ser.CharField()
+    text = ser.CharField()
+
+    class Meta:
+        type_ = 'citations'
+
+    def create(self, validated_data):
+        node = self.context['view'].get_node()
+        errors = self.error_checker(validated_data, node)
+
+        if len(errors) > 0:
+            raise ValidationError(detail=errors)
+        else:
+            citation = AlternativeCitation(text=validated_data['text'], name=validated_data['name'])
+            citation.save()
+            node.alternativeCitations.append(citation)
+            node.save()
+            return citation
+
+    def update(self, instance, validated_data):
+        node = self.context['view'].get_node()
+        errors = self.error_checker(validated_data, node, instance=instance)
+
+        if len(errors) > 0:
+            raise ValidationError(detail=errors)
+        else:
+            instance.name = validated_data.get('name') or instance.name
+            instance.text = validated_data.get('text') or instance.text
+            instance.save()
+            return instance
+
+    def error_checker(self, data, node, instance=None):
+        errors = []
+
+        if 'name' not in data or (instance is not None and instance.name == data['name']):
+            name_exists = False
+        else:
+            name_exists = node.alternativeCitations.find(Q('name', 'eq', data['name'])).count() > 0
```

Comment:
For checking uniqueness of serializer fields, can you possibly use the DRF `UniqueValidator`? http://www.django-rest-framework.org/api-guide/validators/#uniquevalidator